pkg/resource/deploy/step_generator.go

```diff
@@ -112,7 +112,15 @@ func (sg *stepGenerator) Errored() bool {
 // GenerateReadSteps is responsible for producing one or more steps required to service
 // a ReadResourceEvent coming from the language host.
 func (sg *stepGenerator) GenerateReadSteps(event ReadResourceEvent) ([]Step, result.Result) {
+	// Generate a URN for this new resource, confirm we haven't seen it before in this deployment.
 	urn := sg.deployment.generateURN(event.Parent(), event.Type(), event.Name())
+	if sg.urns[urn] {
+		// TODO[pulumi/pulumi-framework#19]: improve this error message!
```

Comment:
That issue is closed, is this the right ref?

Reply:
It's a copy of this code from RegisterResource, I didn't check the issue out. Looks like it's not really fixed, but not sure if we ever will...

Reply:
Lift to a common func to highlight the sharing?

Reply:
Sure